packages/@aws-cdk/aws-ecs/lib/base/base-service.ts

```diff
@@ -466,6 +477,15 @@ export abstract class BaseService extends Resource
       Annotations.of(this).addWarning('taskDefinition and launchType are blanked out when using external deployment controller.');
     }
 
+    // If the circuit breaker chooses not to set the DeploymentController, and it has been manually set to something other than ECS
+    // Add an error that it is not supported
+    if (props.circuitBreaker
```

Comment:
Reading this, I'm wondering if we need to redefine the contract somewhat and deprecate this current version instead of adding in this error.

Reply:
Do you have some guidance/examples on how this would be achieved?